linker/shared/src/main/scala/org/scalajs/linker/frontend/Refiner.scala

```diff
@@ -31,8 +31,10 @@ final class Refiner(config: CommonPhaseConfig, checkIR: Boolean) {
   import Refiner._
 
   private val irLoader = new ClassDefIRLoader
-  private val analyzer =
-    new Analyzer(config, initial = false, checkIR = checkIR, failOnError = true, irLoader)
+  private val analyzer = {
+    val checkIRFor = Some(CheckingPhase.Optimizer).filter(_ => checkIR)
```

Comment:
```suggestion
    val checkIRFor = if (checkIR) Some(CheckingPhase.Optimizer) else None
```

? even if a bit longer, IMO easier to grok.